### PR TITLE
feat: upgraded code-fns for new theme options and lazy loading

### LINKED ISSUE #322: CodeBlock component isn't exported
**Describe the bug**
The `CodeBlock` component isn't exported in [`@motion-canvas/2d/src/components/index.ts`](https://github.com/motion-canvas/motion-canvas/blob/main/packages/2d/src/components/index.ts). Is this intentional?

**To Reproduce**
Attempt to import the CodeBlock component:
`import { CodeBlock } from "@motion-canvas/2d/lib/components";`

**Expected behavior**
The component is successfully imported.

**Console errors**
image

Text form of above error:
`Uncaught SyntaxError: The requested module '/node_modules/.vite/deps/@motion-canvas_2d_lib_components.js?v=c5be1f43' does not provide an export named 'CodeBlock'`

**Package versions:**
- `@motion-canvas/2d: ^2.3.0`

### LINKED ISSUE #396: Upgrade CodeBlock to use Shiki and VSCode themes.
This would be a breaking change, so it would be nice to get it in before 3.0. I'm working on it today.

**Description**
Hunter mentioned Shiki on the discord and I'm now experimenting with it in CodeBlock. It's looking extremely promising. As in, I wish I had known about it before. It would allow for the use of VSCode themes, which would be far better than the theme hack that I did.

If I switch to it it will likely be a breaking change on "theme", since it would switch from an object to a name for a VSCode theme. There might be other changes.

**Proposed solution**
I would update `code-fns`, release a major version, then open a pull for MC.

**Considered alternatives**
I _might_ be able to keep the current theme with another hack and just deprecate it until the 4.0 release. That's what I would do if the 3.0 was years away, but it feels unnecessary at this stage.

In retrospect, I should have brought it in as "experimental", but I never suspected that I would discover a new code highlighter so soon. I searched for a long time before settling on Starry Night.

**Additional context**
If you want your changes in, feel free to release 3.0, but if I can squeeze it in, I'll try to.

### PR DESCRIPTION
**Pull request by Ross-Esmond**
* CodeBlock is now exported
* new stockTheme option with common theme options like github and nord
* new, more granular theme options
* uses the shiki library

closes #396, closes #322

### CODE REVIEW

File: packages/2d/src/components/CodeBlock.ts [hunk 1]
```diff
@@ -47,14 +50,44 @@ export interface CodeModification {
...
   @signal()
-  public declare readonly language: SimpleSignal<string, this>;
+  public declare readonly language: SimpleSignal<Lang, this>;
+
+  @initial(undefined)
+  @signal()
+  public declare readonly stockTheme: SimpleSignal<Theme | undefined, this>;
+
+  protected static loadedLanguages = new Map<Lang, boolean>();
+  protected static loadedThemes = new Map<Theme, boolean>();
```
Comment (aarthificial) [inline packages/2d/src/components/CodeBlock.ts:62]: Looks like these could be simple Sets

File: packages/2d/src/components/CodeBlock.ts [hunk 2]
```diff
@@ -47,14 +50,44 @@ export interface CodeModification {
...
+            CodeBlock.loadedThemes.set(theme, true);
+          }
+        }),
+      );
+      return false;
+    }
+
+    return (
+      CodeBlock.loadedLanguages.get(language) &&
+      (theme == null || CodeBlock.loadedThemes.get(theme))
```
Comment (aarthificial) [inline packages/2d/src/components/CodeBlock.ts:88]: Couldn't it be:
```suggestion
      CodeBlock.loadedLanguages.has(language) &&
      (theme == null || CodeBlock.loadedThemes.has(theme))
```
?

File: packages/2d/src/components/CodeBlock.ts [hunk 3]
```diff
@@ -47,14 +50,44 @@ export interface CodeModification {
...
-  );
-
   @initial('tsx')
   @signal()
-  public declare readonly language: SimpleSignal<string, this>;
+  public declare readonly language: SimpleSignal<Lang, this>;
+
+  @initial(undefined)
+  @signal()
+  public declare readonly stockTheme: SimpleSignal<Theme | undefined, this>;
```
Comment (aarthificial) [inline packages/2d/src/components/CodeBlock.ts:59]: The default value should be `null` to be consistent with other Node properties:
```suggestion
  @initial(null)
  @signal()
  public declare readonly stockTheme: SimpleSignal<Theme | null, this>;
```

File: packages/2d/src/components/CodeBlock.ts (hunk 1 above)
Comment (Ross-Esmond) [reply packages/2d/src/components/CodeBlock.ts:62]: I actually intended for these to use `false` as well. The false means that the language or theme is loading but isn't loaded yet.

File: packages/2d/src/components/CodeBlock.ts (hunk 2 above)
Comment (Ross-Esmond) [reply packages/2d/src/components/CodeBlock.ts:89]: I updated the code.

**Review by aarthificial — APPROVED** [review]

**Pull request merged**